lib/facter/util/resolvers/networking/primary_interface.rb

```diff
@@ -59,6 +59,15 @@ def read_from_ip_route
               primary_interface
             end
 
+            def read_from_ip6_route
+              return if Facter::Core::Execution.which('ip').nil?
+
+              require 'json'
+
+              output = JSON.parse(Facter::Core::Execution.execute('ip -json -6 route show default', logger: @log))
```

Comment:
If the ip command is not available, the result will make JSON.parse throw an exception, which will break the resolver.

Reply:
JSON.parse fails on empty strings `JSON::ParserError (809: unexpected token at '')`, could you please verify if the output of `execute` is not empty before trying to parse it with JSON? This generates the acceptance tests failures.

Reply:
I rebased and tried to verify that the output of `execute` is not empty, but now I don't know what is going on with the failures from `spec/facter/resolvers/networking_linux_spec.rb`.